### ISSUE TITLE
Add r2SCAN-3c and wB97X-3c parameters

### ISSUE THREAD

**Issue by TyBalduf** [dftd4/dftd4#221]
It would be useful to have these as selectable functionals, rather than needing to enter the parameters manually.

**Comment by marvinfriede** [comment]
I added the D4 parameters for both functionals (#220).

However, for r2SCAN-3c, two parameters of the charge-scaling function  ("ga": 2.00, "gc": 1.00) as well as the MBD-scaling ("s9": 2.0) must be modified. These cannot be set internally, but must be specified via the command line interface (`--mbd 2.0 --zeta 2.0 1.0`). Should we just leave that to the user?

**Comment by TyBalduf** [comment]
I think it would be confusing for the end user if specifying `-f r2scan` didn't result in the correct D4 parameters/model.

It seems like `s9` could be modified internally around here: https://github.com/dftd4/dftd4/blob/main/src/dftd4/param.f90#L210 
Essentially, just checking for the functional id for any special cases where `s9` should be modified and setting it before `get_d4eeq_*_parameters` is called.

Updating `ga` and `gc` is a little trickier. If changing the D4 model parameters is something that will be done more often for later functionals, it would maybe make sense to have `new_d4_model` accept a method argument as well. But if we assume this is a one off case, then just checking for the method before the call to `new_d4_model` [in the app](https://github.com/dftd4/dftd4/blob/main/app/driver.f90#L146) and replacing `ga` and `gc` would be enough.

**Comment by marvinfriede** [comment]
I added a check for r2SCAN-3c before `get_rational_damping` collects the damping parameters (including `s9`, so we set it before). This way, we do not break the API of `new_d4_model`, which is not necessary for a single functional in my opinion. The additional code also checks if `--zeta` or `--mbdscale` are set via command line, so they can still be overwritten for r2SCAN-3c, i.e., command line arguments have priority.
